### DISCUSSION TITLE
Prime Tween have a tween like iTween RotateBy ?

### DISCUSSION THREAD

**Discussion by OscarLeif** [KyryloKuzyk/PrimeTween#73] [FAQ] [answered]
Basically I'm not sure if this feature exist.
The goal is to increment the current rotation on the Z axis 
if I say 360 in 1 second the tween should increment the rotation. 
This is very different from like using DoTween Rotate To. it goes to a specific target.

**Comment by KyryloKuzyk** [comment, marked as answer]
You can do this in PrimeTween with Tween.EulerAngles():
```csharp
Tween.EulerAngles(transform, transform.eulerAngles, transform.eulerAngles + new Vector3(0, 0, 360), 1);
```

More info on this topic:
[Getting Started](GenerateRotateByTargets) section
https://docs.unity3d.com/ScriptReference/Transform-eulerAngles.html
https://discussions.unity.com/t/primetween-high-performance-animations-and-sequences/926420/51

**Reply by KyryloKuzyk** [reply]
The reason why PrimeTween doesn't provide the RotateBy() equivalent is that such rotation can't be generalised: the resulting animation depends on the initial rotation.
You can visualise this idea if you apply arbitrary rotation to an object, then drag the x/y/z Rotation property in Inspector. What you'll observe is that only rotation around z axis will produce expected results, but rotation around x/y axis is unpredictable. This is why it's not possible to have the RotateBy() method that will work in all cases. So PrimeTween provides only EulerAngles() method that works exactly the same as setting the transform.eulerAngles property.
